lib/bundles/inspec-compliance/target.rb

```diff
@@ -55,6 +55,9 @@ def self.resolve(target) # rubocop:disable PerceivedComplexity, Metrics/Cyclomat
         end
         profile_fetch_url = Compliance::API.target_url(config, profile)
       end
+      # We need to pass the token to the fetcher
+      config['token'] = Compliance::API.get_token(config)
```

Comment:
Looking at this it might make sense to just follow this convention everywhere.  Just on each request we set the config['token'] so it lives the entire life of the command but then do not save this to disk.